docs/guides/javascript/reactive/index.md

```diff
@@ -0,0 +1,1382 @@
+
+<details>
+  <summary>View example</summary>
+  <div>
+
+```js title="path/to/plugin/amd/src/main.js"
+import {reactiveInstance} from 'YOUR_PLUGIN/reactive';
+import {YourComponent} from 'YOUR_PLUGIN/yourcomponent;
+
+(...)
+
+return new YourComponent({
+     element: document.getElementById(DOM_ELEMENT_ID),
+     reactive: reactiveInstance,
+ (... other data your component needs ...)
+});
+```
+
+  </div>
+</details>
+
+In case no Reactive Instance is passed, the BaseComponent will try to inherit the instance from a parent DOM element. This way subcomponents can omit that param.
+
+### Initialize a component from a mustache template
+
+Components are easy to embed in mustache files. To do so your class must have a static "init" method that could be called inside the {{#JS}}.
+
+<Tabs>
+  <TabItem value="component-view" label="Component class" default>
+
+```js title="path/to/plugin/amd/src/childcomponent.js"
+import {BaseComponent} from 'core/reactive';
+import {reactiveInstance} from 'YOUR_PLUGIN/reactive';
+
+export default class extends BaseComponent {
+    static init(target, selectors) {
+        return new Component({
+            element: document.getElementById(target),
+            reactive: reactiveInstance,
+            selectors,
+        });
+    }
+    // (...)
+}
+```
+
+  </TabItem>
+  <TabItem value="mustache-view" label="Mustache file">
+
+```handlebars title="path/to/plugin/templates/yourthing/childcomponent.mustache"
+<div id="{{uniqid}}-your-plugin-{{id}}"> (...) </div>
+{{#js}}
+require(['YOUR_PLUGIN/local/yourthing/childcomponent'], function(component) {
+    component.init('{{uniqid}}-your-plugin-{{id}}');
+});
+{{/js}}
+```
+
+  </TabItem>
+</Tabs>
+
+The init method has an id as a first param to find the main DOM Element, but it also has a second one called `selectors`. This second param is an optional (but recommended) param any component can use. We will see how to use the selectors later but they are CSS selectors to find inner DOM elements in the main element. Bypassing as an init-param, any mustache can customize the default CSS selectors, which means the component is more reusable.
+
+### The Component lifecycle hooks
+
+All components are registered in a Reactive Instance and, when this is done, the Component can start watching the state and ask for state mutations. However, during its life, a Component can operate differently if necessary.
+
+To understand how a component works you need to understand the component lifecycle and its hook methods:
+
+![Component lifecycle hooks](./_files/lifecycle_hooks.svg)
+
+<Tabs>
+  <TabItem value="create" label="create" default>
+
+Before the component is registered into the Reactive Instance. This method is used to define the main component attributes. At this moment the Component cannot interact with the Reactive Instance yet.
```

Comment:
The first sentence is not a complete sentence. Either turn it into one, or make it a piece of structured data, for example:

```suggestion
Called: Before the component is registered to the Reactive Instance.

This method is used to define the main component attributes. The Component cannot interact with the Reactive Instance yet as instantiation has not yet completed.
```

The above makes it clear where it is called in the lifecycle, and also adds a brief explanation as to why it cannot interact yet.